src/entity/mod.rs

```diff
@@ -35,42 +38,58 @@ pub trait RoundedEntity {
 }
 
 #[derive(Debug)]
-pub struct EntityState {
+pub struct Vector3 {
     pub x: f32,
     pub y: f32,
     pub z: f32,
 }
 
-impl EntityState {
-    pub fn new(x: f32, y: f32, z: f32) -> EntityState {
-        return EntityState { x: x, y: y, z: z };
+impl Vector3 {
+    pub fn new(x: f32, y: f32, z: f32) -> Vector3 {
+        return Vector3 { x: x, y: y, z: z };
     }
 }
 
-impl Clone for EntityState {
-    fn clone(&self) -> EntityState {
-        EntityState {
+impl Clone for Vector3 {
+    fn clone(&self) -> Vector3 {
+        Vector3 {
             x: self.x,
```

Comment:
Do we want to rename this to `PositionVector` instead of `Vector3`? Might be more specific as to what it refers to.